uaclient/tests/test_contract.py

```diff
@@ -1,13 +1,54 @@
+import copy
 import mock
 import pytest
 
 from uaclient.contract import (
-    API_V1_TMPL_CONTEXT_MACHINE_TOKEN_REFRESH, request_updated_contract)
+    API_V1_TMPL_CONTEXT_MACHINE_TOKEN_REFRESH,
+    API_V1_TMPL_RESOURCE_MACHINE_ACCESS, process_entitlement_delta,
+    request_updated_contract)
 
 from uaclient.testing.fakes import FakeConfig, FakeContractClient
 
 
 M_PATH = 'uaclient.contract.'
+M_REPO_PATH = 'uaclient.entitlements.repo.RepoEntitlement.'
+
+
+class TestProcessEntitlementDeltas:
+
+    def test_error_on_missing_entitlement_type(self):
+        """Raise an error when neither dict contains entitlement type."""
+        error_msg = ('Could not determine contract delta service type %s %s' %
+                     ({}, {'something': 'non-empty'}))
+        with pytest.raises(RuntimeError, match=error_msg):
```

Comment:
TIL about the match kwarg to raises, thanks!

Unfortunately, it looks like it's only been in pytest since 3.1.0: https://github.com/pytest-dev/pytest/blob/master/CHANGELOG.rst#310-2017-05-22

This isn't showing up as an error because older versions of `pytest.raises` will silently ignore any kwargs they don't understand (this is on trusty):

```py
>>> import pytest
>>> pytest.raises()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: raises() missing 1 required positional argument: 'ExpectedException'
>>> pytest.raises('foo')
<_pytest.python.RaisesContext object at 0x7fb1765b6eb8>
>>> pytest.raises('foo', bar='baz')
<_pytest.python.RaisesContext object at 0x7fb1755e3128>
```

So we're silently not actually asserting anything about the error message on older versions of pytest.